integration-tests/container/processes-listening-on-ports/plop.py

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env python3
+
+import socket
+
+from flask import Flask
+
+app = Flask(__name__)
+
+
+opened_sockets = {}
+
+
+@app.route('/open/<int:port>')
+def open_port(port: int):
+    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    s.bind((socket.gethostname(), port))
```

Comment:
What happens when this port is already taken? does the error propagate through the HTTP response?

Reply:
It does:
![image](https://github.com/user-attachments/assets/db0f7b27-dae3-4f7d-bef5-00a493822c82)

Reply:
Hmm but the error we'll see will just be the assertion on the response code being 500. It might be worth adding our own error response so we can propagate the precise exception

Reply:
Alright, let me try to catch errors and give meaningful messages back.

Reply:
This should be a bit better
![image](https://github.com/user-attachments/assets/228ccd89-9e95-4bc8-8205-614c954ac03b)